### DISCUSSION TITLE
uart.v 和 uart_rx.v 程式碼請教

### DISCUSSION THREAD

**Discussion by Jacky100011** [bol-edu/HLS-SOC-Discussions#163] [General]
1. 請問在 uart.v 中的這部分，為什麼是用40000000 去除以baud rate ? 

<img width="321" alt="擷取" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/148679493/17295601-d142-4b76-aa04-278ede9dd4d9">

2. 請問在 uart_rx.v 中的這部分，為什麼 clk_div 需要右移一個 bit ?

<img width="477" alt="擷取" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/148679493/bcffa0d4-3979-45e1-8e51-5b570eb40454">

謝謝!

**Comment by Lanxas1018** [comment]
同學好
1. 40M去除以baud rate是因為我們現在是以40MHz去跑模擬，我們需要算出要多少個25ns的clock cycles才能送出或接收1bit的data以對應我們設定的buad rate
2. 上面comment的地方有提到，這是為了要確認我們收到的start bit他並非glitch或其他原因造成的pulse，所以要檢查他在一個uart period中start bit是否維持是0，因此將clk_div除2，等同於shift 1 bit

**Reply by Jacky100011** [reply]
了解，謝謝您!
